internal/stats/metrics.go

```diff
@@ -1,29 +1,169 @@
 package stats
 
 import (
+	"fmt"
+
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/log"
 	"github.com/edulinq/autograder/internal/timestamp"
 )
 
-type Metric interface {
+type MetricAttribute string
+type MetricType string
+
+// Keys for the attributes field inside of Metric and Query.
+const (
+	Unknown_Metric_Attribute_Key MetricAttribute = ""
+	Assignment_ID_Key                            = "assignment"
+	Analysis_Type_Key                            = "analysis-type"
+	Course_ID_Key                                = "course"
+	Endpoint_Key                                 = "endpoint"
+	Locator_Key                                  = "locator"
+	Sender_Key                                   = "sender"
+	Task_Type_Key                                = "task-type"
+	User_Email_Key                               = "user"
+)
+
+// Values for the type field inside of Metric and Query.
+const (
+	Unknown_Metric_Attribute_Type MetricType = ""
+	API_Request_Stats_Type                   = "api-request"
+	Code_Analysis_Time_Stats_Type            = "code-analysis-time"
+	Grading_Time_Stats_Type                  = "grading-time"
+	Task_Time_Stats_Type                     = "task-time"
```

Comment:
We don't use Title_Case for pretty much anything.
Use PascalCase like the user roles:
https://github.com/edulinq/autograder-server/blob/main/internal/model/role_server.go#L17